contribute/design/1.x/cstor/cstor-status.md

```diff
@@ -0,0 +1,220 @@
+
+**Offline** : The device has been explicitly taken offline by the cspi-mgmt controller.
+
+**Unavail** : The device or virtual device cannot be opened. In some cases, pools with UNAVAIL devices appear in DEGRADED mode. If a top-level virtual device is UNAVAIL, then nothing in the pool can be accessed.
+
+**Removed** : The device was physically removed while the system was running. Device removal detection is hardware-dependent and might not be supported on all platforms.
+
+The phase is the current state of the CSPI. With the addition of Conditions to the status we can represent the latest available observations of a CSPI’s current state. The conditions for cspi will be represented by the following structure.
+```go
+// CSPIConditionType describes the state of a CSPI at a certain point.
+type CStorPoolInstanceCondition struct {
+	// Type of CSPI condition.
+	Type CSPIConditionType `json:"type" protobuf:"bytes,1,opt,name=type,casttype=DeploymentConditionType"`
+	// Status of the condition, one of True, False, Unknown.
+	Status corev1.ConditionStatus `json:"status" protobuf:"bytes,2,opt,name=status,casttype=k8s.io/api/core/v1.ConditionStatus"`
+	// The last time this condition's reason (or) message was updated.
+	LastUpdateTime metav1.Time `json:"lastUpdateTime,omitempty" protobuf:"bytes,6,opt,name=lastUpdateTime"`
+	// Last time the condition transitioned from one status to another.
+	LastTransitionTime metav1.Time `json:"lastTransitionTime,omitempty" protobuf:"bytes,7,opt,name=lastTransitionTime"`
+	// The reason for the condition's last transition.
+	Reason string `json:"reason,omitempty" protobuf:"bytes,4,opt,name=reason"`
+	// A human readable message indicating details about the transition.
+	Message string `json:"message,omitempty" protobuf:"bytes,5,opt,name=message"`
+}
+```
+
+When the conditions like expansion or disk replacement is under progress the message and reason fields will get populated with corresponding details and once the condition has reached completion the fields will have some default values as placeholders until the condition is triggered again.
+
+The proposed conditions for CSPI are :
+
+**PodAvailable** : The PodAvailable condition represents whether the CSPI pool pod is running or not. Whenever the PodAvailable is set to False then the CSPI phase should be set to Unavail to tackle the stale phase on the CSPI when the pool pod is not in running state. The owner of this condition will be the CSPC operator as when the pool pod is lost the cspi-mgmt will not be able to update the conditions.
+```yaml
+Conditions:
+  - lastUpdateTime: 
+    lastTransitionTime: 
+    status: "True"
+    type: PodAvailable
+
+Conditions:
+  - lastUpdateTime: 
+    lastTransitionTime: 
+    message: 'pool pod not in running state'
+    reason: MissingPoolDeployment
+    status: "False"
+    type: PodAvailable
+```
+
+**Created** : The Created condition represents whether the pool got created successfully or not.
```

Comment:
I miss this during implementation Do We really need this? Because It will create pool only once in its lifetime(there might be ephemeral cases that we are handling via `PoolLost` so I think we no need this) so no need to have IMO.